src/lib.rs

```diff
@@ -312,18 +312,25 @@ impl Arguments {
             let value = os_to_str(value)?;
             Ok(Some((value, PairKind::TwoArguments, idx)))
         } else if let Some((idx, key)) = self.index_of2(keys) {
-            // Parse a `--key=value` pair.
+            // Parse a `--key=value` or `-Kvalue` pair.
 
             let value = &self.0[idx];
 
             // Only UTF-8 strings are supported in this method.
             let value = value.to_str().ok_or_else(|| Error::NonUtf8Argument)?;
 
             let mut value_range = key.len()..value.len();
-            if value.as_bytes().get(value_range.start) == Some(&b'=') {
+
+            #[cfg(feature = "eq-separator")]
```

Comment:
All the `eq-separator` code can be move into a it's own block, e.g.:

```rust
#[cfg(feature = "eq-separator")]
{
    if value.as_bytes().get(value_range.start) == Some(&b'=') {
        value_range.start += 1;
    } else {
        // Key must be followed by `=`.
        return Err(Error::OptionWithoutAValue(key));
    }
}
```

Reply:
Thanks! I tried this without making it a block and the compiler complained that this wasn't in stable rust yet, but I guess it's because `if` statements can also act as expressions if they're not in their own block.